api/src/main/java/com/datatorrent/api/Module.java

```diff
@@ -36,7 +36,7 @@
  * @since 3.3.0
  */
 @InterfaceStability.Evolving
-public interface Module
+public interface Module extends Operator
```

Comment:
Operator contains execution life cycle methods, which are not appropriate for "Module". Please consider a different design.